<!-- maze.launch: launch stage with a maze_solver node -->

<launch>
  <!-- Run stage node -->
  <node pkg="stage_ros" type="stageros" name="stageros" args="$(find robotcraft_maze)/world/robotcraft_maze.world" />

    <!-- Run gmapping node -->
	  <node pkg="gmapping" type="slam_gmapping" name="gmapping">
  	   <remap from="scan" to="base_scan_0" />
       <!-- Dimensions of scan (values create a square, why?)-->
       <param name="xmin" value="-3.0"/>
       <param name="ymin" value="-6.0"/>
       <param name="xmax" value="3.0"/>
       <param name="ymax" value="6.0"/>
       <!-- Resolution of scan (m/pixel) -->
       <param name="delta" value="0.20"/>
       <!-- Make scans with no obstacle appear as free instead of unkown
       <param name="maxUrange" value="2.95"/>
       <param name="maxRange" value="3.0"/> -->
    </node>

    <!-- Run maze_basic node -->
    <node pkg="robotcraft_maze" type="maze_basic" name="maze_basic" output="screen" />
    <param name="left" type="bool" value="true" />
    <param name="right" type="bool" value="true" />

  <!-- Run map saver node, periodically saves map and robot position backups -->
  <node pkg="robotcraft_maze" type="map_saver" name="map_saver_controller_node" output="screen"/>

</launch>
